SymPEP-0001.md

```diff
@@ -0,0 +1,241 @@
+and developed as SymPEPs. Statements of these policies and goals should be
+incorporated into the SymPy documentation, as determined in the associated
+SymPEP.
+
+Every SymPEP must have at least one champion. The champion(s) are persons who
+are responsible for writing the SymPEP, leading the discussions around it, and
+developing consensus around it.
+
+The author of a SymPEP should fork the repository and create a draft pull
+request with a new SymPEP document based on the [SymPEP
+template](SymPEP-template). The SymPEP document may be named `SymPEP-XXXX.md`
+until a number is assigned. Once the author completes all information in the
+template and would like it to be reviewed, one of the core SymPy developers should
+decide if the proposal is a legtimate proposal and then assign a number to the
+SymPEP, in which case `XXXX` should be replaced with the number with leading
+0s. The pull request can then be moved to an open pull request and is ready for
+community feedback and discussion but remains in draft state. The person who
+assigns the number should also update the
+[README](https://github.com/sympy/SymPEPs/blob/main/README.md) of the main
+[SymPEP repository](https://github.com/sympy/SymPEPs) to list that number.
+Numbers should be assigned to SymPEPs as soon as it is determined that they are
+a legitimate proposal. If a SymPEP is not deemed necessary, the author should
+rework the SymPEP or move it to a normal pull request or issue on the main
+SymPy repository. If a SymPEP ends up being rejected or postponed, it keeps its
+number, as rejection or postponement status is still a discussion that should
+be documented in the SymPEP. SymPEP numbers should generally be assigned in
+increasing numeric order.
+
+Discussion on the SymPEP should then continue on the open pull request.
+Discussions may also take place in other places, such as [GitHub
+discussions](https://github.com/sympy/SymPEPs/discussions) or the [mailing
+list](http://groups.google.com/group/sympy). All discussions should be
+cross-referenced in the "Discussions" section of the SymPEP document.
+
+For each SymPEP, the community should decide whether a draft implementation of
+the change is needed before acceptance or not. For instance, if a SymPEP
+concerns the details of how a feature is implemented, it should be accepted
+before that happens. On the other hand, the community may decide that it cannot
+come to a consensus about a SymPEP until a draft implementation is proposed.
+
+Once the community reaches a consensus about a SymPEP, the status of a SymPEP
+should be updated (see below). This consensus may be to accept or to reject
+the SymPEP, or to defer it. Here the "community" refers to the broader
+community that has a stake in the SymPEP, not just the core SymPy developers.
+The purpose of the SymPEP process is not to create a cabal of decision makers,
+but rather to enhance the involvement of the broader SymPy community in the
+decision making process.
```

Comment:
Here "consensus" is a very tricky concept. Inherently the idea of working by strict consensus does not scale well up to large groups of people because it implies that any single person has a "veto". I often find this tricky on pull requests because often tangential comments might seem to give a negative opinion but it is not clear how strongly that negative opinion is felt.

Also consensus typically refers only to those who participate visibly in a discussion and often there will be others who do not understand the details of something but might still be affected by it.

I can see that the wording here is careful not to define the decision making process formally and perhaps it is better to keep it that way. I suggest softening the word consensus somehow to allow that it does not strictly require unanimous agreement. Perhaps "broad consensus" or "loose consensus" is better?

Reply:
I've used "tacit consensus" in the past. But I think we actually do operate on a proper full consensus, but that the lead developer (Ondrej, then Aaron, and then now possibly you) could overrule a dissenting vote if full consensus can't actually be achieved.

Reply:
Actually, I think it is best to leave it ambiguous. We have a culture of our method of consensus that is publicly visible. If someone attempts to use/push a different way of consensus that is different than we already do, then people will point to the prior approaches to show how things are done. Worst case is that the lead dev takes control and makes a decision outside of consensus.

Reply:
I see the SymPEP process as basically a replacement for the BDFL style decision making we've had in the past (which hasn't really been necessary for a long time anyway). And if we really did want to formalize some sort of actual governance mechanism, that should be done itself through a SymPEP. But that should not be part of this SymPEP, which is just about the template and process for SymPEPs themselves.

Reply:
I'm not sure I see how this is a replacement. Given that we have not defined consensus in an explicit way, this can coexist with BDFL style (which is consensus with fallback on dictator if all else fails). I don't see merging this and accepting it as removing anything about the informal, yet existing, decision making process we already have.

Reply:
Reading through this paragraph again, I agree with Jason. I intentionally left this ambiguous, because the alternative is to outline some rigorous decision making process, which implies some sort of governance which doesn't exist (or at least hasn't been decided on). 

In fact, the main point I was trying to make with this paragraph is that a SymPEP should involve every relevant stakeholder, not just the core dev team (whatever that means). That also includes non-SymPy developers such as users of SymPy and downstream dependents. 

I would use the term "clear consensus". If something isn't clearly agreed upon, then at best that means the default is to not accept, at least without some further decision making process which isn't defined here.